### DISCUSSION TITLE
Max disk size?

### DISCUSSION THREAD

**Discussion by anandskumar** [1000001101000/Debian_on_Buffalo#207] [Q&A] [answered]
I'm trying to install Debian on a LS210D. Right now it maxes out at 4TB, which I'm guessing is due to the MBS partition it uses. With Debian installed, and using GPT, what is the maximum disk size I can squeeze into this NAS? Or is there any harware limitations into play as well?

**Comment by 1000001101000** [comment, marked as answer]
There's a 16TB limit that comes from the 32-bit arm architecture/kernel (2^32 * 4k blocks). I wouldn't be surprised if there were additional overlapping 16TB limitations from the SATA controller/etc.  

Personally, I have not yet owned a disk larger than 8TB.

**Reply by bcoff12** [reply]
I have a 8TB drive running in LS210D, but that is the largest I've tried as well.
